src/client/components/SortMenu.js

```diff
@@ -32,43 +32,67 @@ export class SortMenu extends React.Component {
     this.props.onSortSubmit();
   }
 
+  componentWillMount(){
+    document.addEventListener('mousedown', this.handleClick, false);
+  }
+
+  componentWillUnmount(){
+    document.removeEventListener('mousedown', this.handleClick, false);
+  }
+
+  handleClick = (e) => {
+    console.log(e.target);
+    console.log(this.outer);
+    if (this.outer.contains(e.target)){
+      return;
+    }
+    else{
+      console.log("here");
+      this.handleClickOutside();
+    }
+  }
+
+
 
   render() {
     const { list } = this.props;
     const { listOpen, headerTitle } = this.state;
 
     return (
-      <div className="sortbyList-container">
-        <div className="sortby-header" onClick={() => this.toggleList()}>
-          <div className="sortby-header-title">{headerTitle}</div>
+      <div ref={outer => this.outer = outer}>
+        <div className="sortbyList-container">
+          <div className="sortby-header" onClick={() => this.toggleList()}>
+            <div className="sortby-header-title">{headerTitle}</div>
+          </div>
+          {listOpen
+            && (
+            <ul className="sortbyList">
```

Comment:
Instead of repeating the list element 4 times, I suggest using array.map function to help void the repetition. Something like the followng
```
let listView = !listOpen ? '' : (
<>
list.map( mapping function )
</>
);
```